src/game.cpp

```diff
@@ -10872,24 +10872,45 @@ void game::chat()
     }
 
     uimenu nmenu;
-    nmenu.text = std::string( _("Who do you want to talk to?") );
-
+    nmenu.text = std::string( _( "Who do you want to talk to or yell at?" ) );
+    
     int i = 0;
+
     for( auto &elem : available ) {
         nmenu.addentry( i++, true, MENU_AUTOASSIGN, ( elem )->name );
     }
 
     nmenu.return_invalid = true;
-    nmenu.addentry( i++, true, 'a', _( "Yell" ) );
-    nmenu.addentry( i++, true, 'q', _( "Cancel" ) );
+
+    int yell, yell_sentence, cancel;
+
+    nmenu.addentry( yell = i++, true, 'a', _( "Yell" ) );
```

Comment:
> and it seems that they are constant expression. 

They certainly are. I meant getting rid of any assignments entirely. See [this function](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/src/game.cpp#L7467) as a good example of what I'm talking about. You can use negative values to avoid overlapping with indexes of NPCs:

```c++
    enum class choices {
        just_yell = INT_MIN,
        yell_something,
        ...
    };
```